examples/react-native-todo-list/App.tsx

```diff
@@ -57,9 +70,7 @@ const App = () => {
               updateTypingStatus={(isTyping: boolean) =>
                 updateMyPresence({ isTyping: isTyping })
               }
-              handleOnSubmitEditing={(todo: string) =>
-                todos.push({ text: todo })
-              }
+              handleOnSubmitEditing={(todo: string) => addTodo(todo)}
```

Comment:
Optional, but I think you can improve this example a little bit further. If you move the `isSomeoneTyping` variable and this `updateTypingStatus` callback out of the props of the `TextInputWithButton` component, and use our hooks there directly. This not only reduces necessary prop drilling, it's also better for rendering efficiency, since only the `TextInputWithButton` will have to be rerendered if someone starts/stops typing, instead of the entire `App` component. In general, I think it's helpful to push the use of our hooks down into the component tree as deep as possible as a best practice.

Reply:
Thanks! you are 100% right! I was dealing in my head with the tradeoff performance vs code clarity. 
I decided to centralise all code related to Liveblocks at root component and let all child components  to be self responsible about their behaviours and UI.